lib/improver/utilities/spatial.py

```diff
@@ -0,0 +1,167 @@
+            The name of the coordinate over which the difference
+            have been calculated.
+        diff_along_axis : numpy array
+            Array containing the differences.
+
+        Returns
+        -------
+        diff_cube : Iris.cube.Cube
+            Cube containing the differences calculated along the
+            specified axis.
+        """
+        points = cube.coord(coord_name).points
+        mean_points = (points[1:] + points[:-1]) / 2
+
+        # Copy cube metadata and coordinates into a new cube.
+        # Create a new coordinate for the coordinate along which the
+        # difference has been calculated.
+        metadata_dict = copy.deepcopy(cube.metadata._asdict())
+        diff_cube = Cube(diff_along_axis, **metadata_dict)
+
+        for coord in cube.dim_coords:
+            dims = cube.coord_dims(coord)
+            if coord.name() in [coord_name]:
+                coord = DimCoord(
+                    mean_points, standard_name=coord.standard_name,
+                    units=coord.units)
+            diff_cube.add_dim_coord(coord.copy(), dims)
+        for coord in cube.aux_coords:
+            dims = cube.coord_dims(coord)
+            diff_cube.add_aux_coord(coord.copy(), dims)
+        for coord in cube.derived_coords:
+            dims = cube.coord_dims(coord)
+            diff_cube.add_aux_coord(coord.copy(), dims)
+
+        # Add metadata to indicate that a difference has been calculated.
+        # TODO: update this metadata when proper conventions have been
+        #       agreed upon.
+        cell_method = CellMethod("difference", coords=[coord_name],
+                                 intervals='1 grid length')
+        diff_cube.add_cell_method(cell_method)
+        diff_cube.attributes["form_of_difference"] = (
+            "forward_difference")
+        diff_cube.rename('difference_of_' + cube.name())
+        print diff_cube
+        return diff_cube
+
+    def calculate_difference(self, cube, coord_axis):
+        """
+        Calculate the difference along the axis specified by the
+        coordinate.
+
+        Parameters
+        ----------
+        cube : Iris.cube.Cube
+            Cube from which the differences will be calculated.
+        coord_axis : String
+            Short-hand reference for the x or y coordinate, as allowed by
+            iris.util.guess_coord_axis.
+
+        Returns
+        -------
+        diff_cube : Iris.cube.Cube
+            Cube after the differences have been calculated along the
+            specified axis.
+        """
+        coord_name = cube.coord(axis=coord_axis).name()
+        diff_axis = cube.coord_dims(coord_name)[0]
+        diff_along_axis = np.diff(cube.data, axis=diff_axis)
+        diff_cube = self.create_difference_cube(
+            cube, coord_name, diff_along_axis)
+        return diff_cube
+
+    def process(self, cube):
+        """
+        Calculate the difference along the x and y axes and return
```

Comment:
I think it might be worth saying it uses np.diff to calculate the differences so that people can go and look up the behaviour at edges etc.